libweb3jsonrpc/WebThreeStubServerBase.cpp

```diff
@@ -124,35 +145,30 @@ static dev::eth::LogFilter toLogFilter(Json::Value const& _json)	// commented to
 	if (!_json.isObject() || _json.empty())
 		return filter;
 
-	if (_json["earliest"].isInt())
-		filter.withEarliest(_json["earliest"].asInt());
-	if (_json["latest"].isInt())
-		filter.withLatest(_json["lastest"].asInt());
-	if (_json["max"].isInt())
-		filter.withMax(_json["max"].asInt());
-	if (_json["skip"].isInt())
-		filter.withSkip(_json["skip"].asInt());
+	// check only !empty. it should throw exceptions if input params are incorrect
+	if (!_json["fromBlock"].empty())
+		filter.withEarliest(jsToInt(_json["fromBlock"].asString()));
+	if (!_json["toBlock"].empty())
+		filter.withLatest(jsToInt(_json["toBlock"].asString()));
+	if (!_json["limit"].empty())
+		filter.withMax(jsToInt(_json["limit"].asString()));
+	if (!_json["offset"].empty())
+		filter.withSkip(jsToInt(_json["offset"].asString()));
 	if (!_json["address"].empty())
 	{
 		if (_json["address"].isArray())
-		{
 			for (auto i : _json["address"])
-				if (i.isString())
-					filter.address(jsToAddress(i.asString()));
-		}
-		else if (_json["address"].isString())
+				filter.address(jsToAddress(i.asString()));
+		else
 			filter.address(jsToAddress(_json["address"].asString()));
 	}
-	if (!_json["topic"].empty() && _json["topic"].isArray())
+	if (!_json["topics"].empty())
 	{
 		unsigned i = 0;
-		for (auto t: _json["topic"])
+		for (auto t: _json["topics"])
 		{
-			if (t.isArray())
-				for (auto tt: t)
-					filter.topic(i, jsToFixed<32>(tt.asString()));
-			else if (t.isString())
-				filter.topic(i, jsToFixed<32>(t.asString()));
+			for (auto tt: t)
+				filter.topic(i, jsToFixed<32>(tt.asString()));
 			i++;
```

Comment:
The result of this is never used. This line, along with the outer-for and outer-if curlies, and 'unsigned i = 0', can be removed.